orcid_service/views.py

```diff
@@ -70,25 +70,55 @@ def orcid_profile(orcid_id):
 
 
 @advertise(scopes=[], rate_limit = [1000, 3600*24])
-@bp.route('/<orcid_id>/orcid-works', methods=['GET', 'POST', 'PUT'])
-def orcid_works(orcid_id):
+@bp.route('/<orcid_id>/orcid-works/<putcode>', methods=['GET', 'PUT', 'DELETE'])
+def orcid_works(orcid_id,putcode):
     '''Get/Set /[orcid-id]/orcid-works - all communication exclusively in JSON'''
 
     payload, headers = check_request(request)
 
     if request.method == 'GET':
-        r = requests.get(current_app.config['ORCID_API_ENDPOINT'] + '/' + orcid_id + '/works',
-                      headers=headers)
+        if ',' in putcode:
+            r = requests.get(current_app.config['ORCID_API_ENDPOINT'] + '/' + orcid_id + '/works/' + putcode,
+                             headers=headers)
+        else:
+            r = requests.get(current_app.config['ORCID_API_ENDPOINT'] + '/' + orcid_id + '/work/' + putcode,
+                          headers=headers)
     elif request.method == 'PUT':
-        r = requests.put(current_app.config['ORCID_API_ENDPOINT'] + '/' + orcid_id + '/works',
+        r = requests.put(current_app.config['ORCID_API_ENDPOINT'] + '/' + orcid_id + '/work/' + putcode,
                       json=payload, headers=headers)
         update_profile(orcid_id)
-    elif request.method == 'POST':
-        r = requests.post(current_app.config['ORCID_API_ENDPOINT'] + '/' + orcid_id + '/works',
-                      json=payload, headers=headers)
+    elif request.method == 'DELETE':
+        r = requests.delete(current_app.config['ORCID_API_ENDPOINT'] + '/' + orcid_id + '/work/' + putcode,
+                      headers=headers)
         update_profile(orcid_id)
-        
-        
+
+    return r.text, r.status_code
+
+
+@advertise(scopes=[], rate_limit=[1000, 3600 * 24])
+@bp.route('/<orcid_id>/orcid-work', methods=['POST'])
```

Comment:
I'm assuming that we are mirroring the orcid api endpoints here (which is fine). But just to check, do they (ie. orcid) really have two enpoints? 

```
/<orcid-id>/orcid-work
/<orcid-id>/orcid-works
```

I don't really like that but we are probably not going to correct that (in our api microservice). So if you say 'yes, they do' I'll merge - I like the rest.

Reply:
Yes we have two endpoint. Bulk and single item.

Reply:
Thanks @rcpeters, we got it directly from the source :)